optimum/habana/transformers/generation/utils.py

```diff
@@ -480,16 +622,14 @@ def generate(
         )
 
         # 9. prepare stopping criteria
-        stopping_criteria = self._get_stopping_criteria(
-            generation_config=generation_config, stopping_criteria=stopping_criteria
-        )
-        if "token_idx" in model_kwargs:
-            if generation_config.max_new_tokens is not None:
-                stopping_criteria.append(StaticMaxLengthCriteria(generation_config.max_new_tokens))
-            else:
-                raise ValueError(
-                    "You need to set `max_new_tokens` in your generation configuration to use static shapes."
-                )
+        if self.config.is_encoder_decoder and generation_mode == GenerationMode.GREEDY_SEARCH:
+            stopping_criteria = _get_stopping_criteria(
+                self, generation_config=generation_config, stopping_criteria=stopping_criteria
+            )
+        else:
+            stopping_criteria = self._get_stopping_criteria(
+                generation_config=generation_config, stopping_criteria=stopping_criteria
+            )
```

Comment:
We should not have two different methods with the same name. All custom logic should be implemented in `self._get_stopping_criteria`.

Reply:
I will move the custom logic to _gaudi_get_stopping_criteria and add it to GaudiGenerationMixin